src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -575,25 +563,31 @@ static int add_partition_esp(DissectedPartition *p, bool has_xbootldr) {
          * Otherwise, we do nothing. */
         if (!has_xbootldr && slash_boot_exists()) {
                 r = slash_boot_in_fstab();
+                if (r < 0)
+                        return r;
+                if (r > 0)
+                        return 0;
```

Comment:
Why?? If XBOOTLDR is mounted to /boot/, we definitely want to mount ESP to /efi/ if possible. If the device node for ESP is listed in fstab, it would be skipped. See devnode_same call above.